scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/synchronization/AOSynchronization.java

```diff
@@ -0,0 +1,824 @@
+            commitIfNeeded(channel);
+            logWithContext(taskid,
+                           channel,
+                           "Run computeIfPresent on key '" + key + "', result is: " + answer,
+                           Level.DEBUG);
+        } catch (InvalidChannelException | CompilationException | IOException e) {
+            throw e;
+        } catch (Throwable e) {
+            throw new ClosureEvaluationException("Exception when evaluating closure : " +
+                                                 StackTraceUtil.getStackTrace(e));
+        }
+        return answer;
+    }
+
+    @Override
+    public <T extends Serializable> T compute(String originator, TaskId taskid, String channel, String key,
+            String remappingFunction) throws InvalidChannelException, CompilationException, IOException {
+        T answer;
+        try {
+            answer = ((Channel<T>) getChannel(channel)).compute(key,
+                                                                (BiFunction<? super String, ? super T, ? extends T>) evaluateClosure(remappingFunction,
+                                                                                                                                     BiFunction.class));
+            commitIfNeeded(channel);
+            logWithContext(taskid, channel, "Run compute on key '" + key + "', result is: " + answer, Level.DEBUG);
+        } catch (InvalidChannelException | CompilationException | IOException e) {
+            throw e;
+        } catch (Throwable e) {
+            throw new ClosureEvaluationException("Exception when evaluating closure : " +
+                                                 StackTraceUtil.getStackTrace(e));
+        }
+
+        return answer;
+    }
+
+    @Override
+    public <T extends Serializable> T merge(String originator, TaskId taskid, String channel, String key, T value,
+            String remappingFunction) throws InvalidChannelException, CompilationException, IOException {
+        T answer;
+        try {
+            answer = ((Channel<T>) getChannel(channel)).merge(key,
+                                                              value,
+                                                              (BiFunction<? super T, ? super T, ? extends T>) evaluateClosure(remappingFunction,
+                                                                                                                              BiFunction.class));
+            commitIfNeeded(channel);
+            logWithContext(taskid,
+                           channel,
+                           "Run merge on key '" + key + "', with value " + value + " result is: " + answer,
+                           Level.DEBUG);
+        } catch (InvalidChannelException | CompilationException | IOException e) {
+            throw e;
+        } catch (Throwable e) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Catch Exception instead of Throwable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1181)